### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/snansumpw`

### PR DESCRIPTION
**Pull request by Snehil-Shah**
## Description

> What is the purpose of this pull request?

This pull request:

- adds C `ndarray` API for `blas/ext/base/snansumpw`.
- refactors package to follow updated project conventions.
- Updates stride handling where stride is equal to zero.
- Adds test cases for `NaN` values.

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/snansumpw/lib/ndarray.js [hunk 1]
```diff
@@ -76,61 +76,61 @@ function snansumpw( N, x, stride, offset ) {
 	if ( N <= 0 ) {
 		return 0.0;
 	}
-	if ( N === 1 || stride === 0 ) {
-		if ( isnanf( x[ offset ] ) ) {
+	ix = offsetX;
+	if ( strideX === 0 ) {
+		if ( isnanf( x[ offsetX ] ) ) {
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumpw/lib/ndarray.js]:
```suggestion
		if ( isnanf( x[ ix ] ) ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/addon.c [hunk 2]
```diff
@@ -34,14 +35,27 @@
...
-    STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 1 );
-
-    napi_value v;
-    napi_status status = napi_create_double( env, stdlib_strided_snansumpw( N, X, stride ), &v );
-    assert( status == napi_ok );
+    STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+    STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snansumpw)( N, X, strideX ), v )
+    return v;
+}
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/addon.c]:
```suggestion
    STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
    STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
    STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snansumpw)( N, X, strideX ), v );
    return v;
}
```

File: lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/addon.c [hunk 3]
```diff
@@ -34,14 +35,27 @@
...
+*/
+static napi_value addon_method( napi_env env, napi_callback_info info ) {
+    STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+    STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+    STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+    STDLIB_NAPI_ARGV_INT64( env, offsetX, argv, 3 );
+    STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snansumpw_ndarray)( N, X, strideX, offsetX ), v )
     return v;
 }
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/addon.c]:
```suggestion
    STDLIB_NAPI_ARGV_INT64( env, offsetX, argv, 3 );
    STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
    STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snansumpw_ndarray)( N, X, strideX, offsetX ), v )
    return v;
}
```

File: lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/main.c [hunk 4]
```diff
@@ -56,16 +77,12 @@ float stdlib_strided_snansumpw( const int64_t N, const float *X, const int64_t s
 	if ( N <= 0 ) {
 		return 0.0f;
 	}
-	if ( N == 1 || stride == 0 ) {
+	ix = offsetX;
+	if ( strideX == 0 ) {
 		if ( stdlib_base_is_nanf( X[ 0 ] ) ) {
 			return 0.0f;
 		}
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/main.c]:
```suggestion
	if ( strideX == 0 ) {
		if ( stdlib_base_is_nanf( X[ ix ] ) ) {
			return 0.0f;
		}
```

Comment (headlessNode) [reply lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/main.c]: Consider this case:
```
stdlib_strided_snansumpw_ndarray(  4, [ 1.0f, 0.0f/0.0f, 2.0f, 3.0f ], 0, 1 );
```
For this case, the current implementation will pass the `nanCheck` and return the value of ` N * X[ ix ]` which in this case would be `4 * X[ 1 ]` => `4 * (0.0f/0.0f)`

File: lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/main.c [hunk 5]
```diff
@@ -74,48 +91,48 @@ float stdlib_strided_snansumpw( const int64_t N, const float *X, const int64_t s
...
+			ix += strideX;
 			s6 += ( stdlib_base_is_nanf( X[ ix ] ) ) ? 0.0 : X[ ix ];
-			ix += stride;
+			ix += strideX;
 			s7 += ( stdlib_base_is_nanf( X[ ix ] ) ) ? 0.0 : X[ ix ];
-			ix += stride;
+			ix += strideX;
 		}
 		// Pairwise sum the accumulators:
 		sum = ((s0+s1) + (s2+s3)) + ((s4+s5) + (s6+s7));
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumpw/src/main.c]:
```suggestion
		sum = ( (s0+s1) + (s2+s3) ) + ( (s4+s5) + (s6+s7) );
```

File: lib/node_modules/@stdlib/blas/ext/base/snansumpw/test/test.snansumpw.js [hunk 6]
```diff
@@ -34,7 +34,7 @@ tape( 'main export is a function', function test( t ) {
 });
 
 tape( 'the function has an arity of 3', function test( t ) {
-	t.strictEqual( snansumpw.length, 3, 'reutrn expected value' );
+	t.strictEqual( snansumpw.length, 3, 'return expected value' );
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumpw/test/test.snansumpw.js]: Thanks for catching this.

**Review by headlessNode — COMMENTED** [review]
@Snehil-Shah Left an initial round of review. Thanks for working on this.

**Review by Planeshifter — APPROVED** [review]
Looks good to me! Thanks @Snehil-Sha for your PR.

**Pull request merged**